src/test/rgw/rgw_multi/tests.py

```diff
@@ -1314,3 +1315,730 @@ def test_bucket_creation_time():
+    get_sync_policy(c1)
+
+    buckets = []
+    content = 'asdasd'
+
+    # create bucketA & objects in zoneA
+    objnamesA = [ 'obj1', 'obj2', 'obj3' ]
+    bucketA = create_zone_bucket(zcA)
+    buckets.append(bucketA)
+    create_objects(zcA, bucketA, objnamesA, content)
+
+    # create bucketB & objects in zoneB
+    objnamesB = [ 'obj4', 'obj5', 'obj6' ]
+    bucketB = create_zone_bucket(zcB)
+    buckets.append(bucketB)
+    create_objects(zcB, bucketB, objnamesB, content)
+
+    zonegroup_meta_checkpoint(zonegroup)
+    zone_data_checkpoint(zoneB, zoneA)
+
+    #for b in buckets:
+    #    log.info('array contains bucket=%s', b.name)
+
+    # verify if objnamesA synced to only zoneB but not zoneC
+    bucket = get_bucket(zcB, bucketA.name)
+    check_objects_exist(bucket, objnamesA, content)
+
+    bucket = get_bucket(zcC, bucketA.name)
+    check_objects_not_exist(bucket, objnamesA)
+
+    # verify if objnamesB are not synced to either zoneA or zoneC
+    bucket = get_bucket(zcA, bucketB.name)
+    check_objects_not_exist(bucket, objnamesB)
+
+    bucket = get_bucket(zcC, bucketB.name)
+    check_objects_not_exist(bucket, objnamesB)
+
+    return
+
+def test_sync_pipe_single_bucket():
+    """
+    symmetrical flow but configure pipe
+    for only bucketA.
+
+    verify only bucketA syncs but not other buckets.
+    """
+
+    zonegroup = realm.master_zonegroup()
+    zonegroup_meta_checkpoint(zonegroup)
+
+    zonegroup_conns = ZonegroupConns(zonegroup)
+
+    (zoneA, zoneB) = zonegroup.zones[0:2]
+    (zcA, zcB) = zonegroup_conns.zones[0:2]
+
+    c1 = zoneA.cluster
+
+    c1.admin(['sync policy get'])
+
+    zones = zoneA.name + ',' + zoneB.name
+    create_sync_policy_group(c1, "sync-group")
+    create_sync_group_flow_symmetrical(c1, "sync-group", "sync-flow1", zones)
+    #create_sync_group_flow_directional(c1, "sync-group", "sync-flow", zoneA.name, zoneB.name)
+
+    zonegroup.period.update(zoneA, commit=True)
+    get_sync_policy(c1)
+
+    objnames = [ 'obj1', 'obj2' ]
+    content = 'asdasd'
+    buckets = []
+
+    # create bucket in zoneA
+    bucketA = create_zone_bucket(zcA)
+    buckets.append(bucketA)
+
+    # configure pipe for only bucketA
+    args = "--source-bucket " + bucketA.name + " --dest-bucket " + bucketA.name
+
+    create_sync_group_pipe(c1, "sync-group", "sync-pipe", zones, zones, None, args)
+    #create_sync_group_pipe(c1, "sync-group", "sync-pipe", zoneA.name, zoneB.name, None, args)
+    set_sync_policy_group_status(c1, "sync-group", "enabled")
+    get_sync_policy(c1)
+    zonegroup.period.update(zoneA, commit=True)
+
+    zoneA.start()
+    zoneB.start()
+
+    # create object in bucketA
+    create_object(zcA, bucketA, objnames[0], content)
+
+    # create another bucket and objects in it
+    bucketB = create_zone_bucket(zcA)
+    buckets.append(bucketB)
+    create_object(zcA, bucketB, objnames[1], content)
+
+    zonegroup_meta_checkpoint(zonegroup)
+    zone_data_checkpoint(zoneB, zoneA)
```

Comment:
For all other testcases above (default pipe configuration), 'zone_data_checkpoint' was enough to wait for sync to finish. 
But for this testcase, sync dint happen as expected.  So I have added zone_bucket_checkpoint too to ensure sync finished. But thats resulting in a seg fault of the command.

Reply:
When tested manually, had to restart the radosgw processes for the sync policy to get applied properly. I tried the same here (line:1772) but that dint work as well.